### DISCUSSION TITLE
openXR input support

### DISCUSSION THREAD

**Discussion by BlueScorpioVR** [opentrack/opentrack#1437] [Q&A]
Any chance that OpenXR can be added as an input? This is to support WMR headsets like the HP reverb G2 without having to run steam VR. I am planning to use it in MSFS 2020 and there is a noticeable performance drop when using steamVR over OpenXR directly.

**Comment by drowhunter** [comment]
+1

**Comment by M0nk3yofdoom** [comment]
So i'm investigating a way to do this but am struggling. There is a program called OpenXR Motion Compensation https://github.com/BuzzteeBear/OpenXR-MotionCompensation that creates the API layer that captures the head position of the headset in OpenXR. I'm guessing that using this API layer with a tracker plugin for OpenTrack should theoretically be possible to unlock that headset tracking with OpenXR. Any thoughts and guidance on doing this?

**Reply by drowhunter** [reply]
Funny you should ask.

This fork has OpenXR tracking as well as OpenVR support

https://github.com/Ofisare/FreePIE

**Reply by M0nk3yofdoom** [reply]
This looks like it tracks hands in OpenXR but not head position? Or did I miss how that can be done? I want to assign head position to movement controls.

**Reply by wotupfoo** [reply]
> So i'm investigating a way to do this but am struggling. There is a program called OpenXR Motion Compensation https://github.com/BuzzteeBear/OpenXR-MotionCompensation that creates the API layer that captures the head position of the headset in OpenXR. I'm guessing that using this API layer with a tracker plugin for OpenTrack should theoretically be possible to unlock that headset tracking with OpenXR. Any thoughts and guidance on doing this?

Did you ever get around to working on this?
I'd be interested.

**Reply by M0nk3yofdoom** [reply]
I did not make any progress. I'm looking at some UEVR profiles that incorporate motion tracking and head location. I'll be attempting to unwrap some of those to help make them act like an axis.

**Reply by M0nk3yofdoom** [reply]
This might help a little bit if you're looking at the UEVR route:
https://www.nexusmods.com/mxgppro/mods/1?tab=files

And a video on what it can do. 
https://www.youtube.com/watch?v=_z41V9uFBRs
